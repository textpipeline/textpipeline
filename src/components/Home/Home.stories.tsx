import Box from '@material-ui/core/Box';
import { storiesOf } from '@storybook/react';
import React from 'react';

import Home from './Home';

storiesOf('Home', module).add('default', () => (
  <Box height="50vh">
    <Home />
  </Box>
));
