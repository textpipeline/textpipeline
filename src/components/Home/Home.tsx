import Box from '@material-ui/core/Box';
import { createStyles, makeStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import RepeatIcon from '@material-ui/icons/Repeat';
import Heading from 'components/Heading';
import React from 'react';

const useStyles = makeStyles(() =>
  createStyles({
    icon: {
      fontSize: 150,
    },
  })
);

const Home: React.FC = () => {
  const classes = useStyles();

  return (
    <Box width="100%" height="100%" display="flex" flexDirection="column" justifyContent="center" alignItems="center">
      <Heading level={1} semanticLevel={2}>
        What text do you want to transform?
      </Heading>

      <Typography variant="subtitle1">Choose a text transform from the menu.</Typography>

      <RepeatIcon className={classes.icon} />
    </Box>
  );
};

export default Home;
