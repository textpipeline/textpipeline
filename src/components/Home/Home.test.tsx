import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

import Home from './Home';

describe('Home', () => {
  let rendered: ShallowWrapper;

  beforeEach(() => {
    rendered = shallow(<Home />);
  });

  it('renders correctly', () => {
    expect(rendered).toMatchSnapshot();
  });
});
