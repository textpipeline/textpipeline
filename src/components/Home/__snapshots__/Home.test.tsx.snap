// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<Styled(MuiBox)
  alignItems="center"
  display="flex"
  flexDirection="column"
  height="100%"
  justifyContent="center"
  width="100%"
>
  <Heading
    level={1}
    semanticLevel={2}
  >
    What text do you want to transform?
  </Heading>
  <WithStyles(ForwardRef(Typography))
    variant="subtitle1"
  >
    Choose a text transform from the menu.
  </WithStyles(ForwardRef(Typography))>
  <RepeatIcon
    className="makeStyles-icon-1"
  />
</Styled(MuiBox)>
`;
